File: day32.py
#User function Template for python3

class Solution:
    def largestPrimeFactor (self, N):
        # code here
        factor=0
        d=2
        while d*d<=N:
            while N%d==0:
                factor=max(factor,d)
                N//=d
            d+=1
        if N>1:
            factor=max(factor,N)
        return factor    
            


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        N = int(input())
       

        ob = Solution()
        print(ob.largestPrimeFactor(N))
# } Driver Code Ends

File: day32.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std;

// } Driver Code Ends

class Solution{
public: 
    long long int largestPrimeFactor(int N){
       long long int factor=0,d=2;
       //int d=2;
        while (d*d<=N)
        {
            while(N%d==0){
            factor=d;
            N/=d;
            }
            d+=1;
        }
        if(N>1)
        factor=N;

        return factor;

    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N;
        cin>>N;
        Solution ob;
        cout << ob.largestPrimeFactor(N) << endl;
    }
    return 0; 
}
// } Driver Code Ends